src/components/DataViewWrapper.tsx

```diff
@@ -0,0 +1,374 @@
+import { EmptyState, EmptyStateBody, Pagination } from "@patternfly/react-core";
+import { DataView } from "@patternfly/react-data-view/dist/dynamic/DataView";
+import { DataViewFilters } from "@patternfly/react-data-view/dist/dynamic/DataViewFilters";
+import {
+  DataViewTable,
+  DataViewTh,
+} from "@patternfly/react-data-view/dist/dynamic/DataViewTable";
+import { DataViewTextFilter } from "@patternfly/react-data-view/dist/dynamic/DataViewTextFilter";
+import { DataViewToolbar } from "@patternfly/react-data-view/dist/dynamic/DataViewToolbar";
+import {
+  useDataViewFilters,
+  useDataViewPagination,
+  useDataViewSort,
+} from "@patternfly/react-data-view/dist/dynamic/Hooks";
+import { CubesIcon } from "@patternfly/react-icons";
+import { Tbody, Td, ThProps, Tr } from "@patternfly/react-table";
+import { FunctionComponent, useMemo } from "react";
+
+import ErrorPlaceholder from "./ErrorPlaceholder";
+
+interface FieldData {
+  name: string;
+  data_path: string;
+  data: (string | number | boolean | null | (string | number)[])[];
+}
+
+interface DataViewColumn {
+  label: string;
+  key: string;
+  sortable?: boolean;
+  filterable?: boolean;
+}
+
+export interface DataViewWrapperProps {
+  component: "data-view" | "table"; // "table" supported for backwards compatibility
+  title?: string;
+  id: string;
+  fields: FieldData[];
+  className?: string;
+  columns?: DataViewColumn[];
+  perPage?: number;
+  enableFilters?: boolean; // If undefined, auto-disables when 5 or fewer items
+  enablePagination?: boolean; // If undefined, auto-disables when 5 or fewer items
+  enableSort?: boolean;
+  emptyStateMessage?: string;
+}
+
+const perPageOptions = [
+  { title: "5", value: 5 },
+  { title: "10", value: 10 },
+  { title: "20", value: 20 },
+  { title: "50", value: 50 },
+];
+
+const DataViewWrapper: FunctionComponent<DataViewWrapperProps> = ({
+  id,
+  fields,
+  className,
+  perPage: initialPerPage = 5,
+  enableFilters,
+  enablePagination,
+  enableSort = true,
+  emptyStateMessage = "No data available",
+}) => {
+  // Check for missing or invalid data
+  const hasNoFields = !fields || fields.length === 0;
+
+  // Transform fields data into table format
+  const { columns, rows, filterableFields } = useMemo(() => {
+    if (hasNoFields) {
+      return { columns: [], rows: [], filterableFields: [] };
+    }
+
+    // Find the maximum number of data items across all fields
+    const maxDataLength = Math.max(...fields.map((field) => field.data.length));
+
+    // Create columns from field names
+    const transformedColumns: DataViewColumn[] = fields.map((field) => ({
+      key: field.name,
+      label: field.name,
+      sortable: true,
+      filterable: true,
+    }));
+
+    // Helper function to format ISO dates for display
+    const formatValue = (
```

Comment:
Could this method be extracted to some reusable form (separate source file) and reused in `one-card` and `set-of-cards` components also?